app/models/course.rb

```diff
@@ -1,3 +1,13 @@
 class Course < Base
   belongs_to :provider
```

Comment:
If we make this declaration:

```
belongs_to :provider, param: :provider_code
```

We'll be able to use it like this:

```
Course.where(provider_code: params[:provider_code]).find(...)
```

Which is a bit nicer than using the misnomer `id`.